File: frontends/configurator/www/model/config/layout_type.js
export const LayoutType = Object.freeze({
	SINGLE: {
		label: 'Single',
		repeatable: false,
		has_dataset_model: false
	},
	MULTIPLE: {
		label: 'Multiple',
		repeatable: true,
		has_dataset_model: true
	}
});

File: backend/src/main/java/ch/rodano/configuration/model/layout/LayoutType.java
package ch.rodano.configuration.model.layout;

public enum LayoutType {
	SINGLE {
		@Override
		public boolean isRepeatable() {
			return false;
		}

		@Override
		public boolean hasReferenceDataset() {
			return false;
		}
	},
	MULTIPLE {
		@Override
		public boolean isRepeatable() {
			return true;
		}

		@Override
		public boolean hasReferenceDataset() {
			return true;
		}
	};

	public abstract boolean isRepeatable();

	public abstract boolean hasReferenceDataset();
}
